backend/app/server/server.go

```diff
@@ -86,7 +90,24 @@ func (s Server) routes() chi.Router {
 		render.PlainText(w, r, "User-agent: *\nDisallow: /api/\nDisallow: /show/\n")
 	})
 
-	s.fileServer(router, "/", http.Dir(s.WebRoot))
+	router.NotFound(func(w http.ResponseWriter, r *http.Request) {
+		if strings.HasPrefix(r.URL.Path, "/api/v1") {
+			render.Status(r, http.StatusNotFound)
+			render.JSON(w, r, JSON{"error": "not found"})
+			return
+		}
+
+		s.render(w, http.StatusNotFound, "404.tmpl.html", baseTmpl, "not found")
+	})
+
+	router.Get("/", s.indexCtrl)
+	router.Post("/generate-link", s.generateLink)
```

Comment:
adding Ctrl suffix to all the routed method will help to see them right away and will keep naming consistent